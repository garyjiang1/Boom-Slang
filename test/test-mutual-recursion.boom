def is_even(int n) returns boolean:
	if n == 0:
		return true
	else:
		return is_odd(n - 1)

def is_odd(int n) returns boolean:
	if n == 0:
		return false
	else:
		return is_even(n - 1)

int[] arr = [3,16,27,35]
int i = 0
loop i+=1 while i < len(arr):
	int elem = arr[i]
	println(elem + " is even? " + is_even(elem))
	println(elem + " is odd? " + is_odd(elem))
