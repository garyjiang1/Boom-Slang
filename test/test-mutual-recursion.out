3 is even? false
3 is odd? true
16 is even? true
16 is odd? false
27 is even? false
27 is odd? true
35 is even? false
35 is odd? true
